pkg/skaffold/walk/walk.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package walk
+
+import (
+	"os"
+
+	"github.com/karrick/godirwalk"
+)
+
+// Dirent stores the name and type of a file system entry.
+type Dirent interface {
+	IsDir() bool
+	Name() string
+}
+
+// Predicate represents a predicate on file system entries.
+// Given a file's absolute path and information, it returns `true`
+// when the predicate is matched. It can also return a `filepath.SkipDir`
+// error to skip a directory and its children altogether.
+type Predicate func(absPath string, info Dirent) (bool, error)
+
+// Action is a function that takes a file's absolute path and information,
+// and optionally returns an error.
+type Action func(absPath string, info Dirent) error
```

Comment:
Confirmed that `godirwalk.Walk()` doesn't return absolute files unless an absolute directory was passed in.  So these should just be  `path` and skip any mention of absolute.